hawkmoth/__init__.py

```diff
@@ -250,12 +250,44 @@ class CAutoEnumDirective(_AutoCompoundDirective):
     _domain = 'c'
     _docstring_types = [docstring.EnumDocstring]
 
+class CppAutoDocDirective(_AutoDocDirective):
+    _domain = 'cpp'
+
+class CppAutoVarDirective(_AutoSymbolDirective):
+    _domain = 'cpp'
+    _docstring_types = [docstring.VarDocstring]
+
+class CppAutoTypeDirective(_AutoSymbolDirective):
+    _domain = 'cpp'
+    _docstring_types = [docstring.TypeDocstring]
+
+class CppAutoMacroDirective(_AutoSymbolDirective):
+    _domain = 'cpp'
+    _docstring_types = [docstring.MacroDocstring, docstring.MacroFunctionDocstring]
+
```

Comment:
Hmm, I hadn't considered not having a `cpp:automacro`... I can see the reason not to, even if I think that it's Sphinx's shortcoming for not having `cpp:macro`.

The issue then becomes that the `c:automacro` directive needs to _not_ impose the domain and cannot check the domain for consistency against the Clang interpretation of the file. This would be a net loss I think, and involve substantial changes to the approach so far.

Also, `cpp:autodoc` should still produce `c:macro`, right? So we could argue the lines are blurry already and we're not making it any worse. Thoughts?